dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/meter/C3MeterChartDisplayer.java

```diff
@@ -131,26 +135,42 @@ private C3Gauge createGauge() {
 
     String[][] extractGroupingValues(DataColumn groupsColumn, DataColumn valuesColumn) {
         int n = groupsColumn.getValues().size();
+        ColumnSettings settings = displayerSettings.getColumnSettings(valuesColumn);
         String[][] output = new String[n][2];
         for (int i = 0; i < n; i++) {
             Object group = groupsColumn.getValues().get(i);
             Object value = valuesColumn.getValues().get(i);
             output[i][0] = columnValueToString(group);
-            output[i][1] = columnValueToString(value);
+            output[i][1] = evaluateValueToString(value, settings);
         }
         return output;
     }
 
     String[] extractSingleColumnValues(DataColumn dataColumn) {
         List<?> values = dataColumn.getValues();
+        ColumnSettings settings = displayerSettings.getColumnSettings(dataColumn);
         String[] data = new String[values.size() + 1];
         data[0] = dataColumn.getId();
         for (int i = 0; i < values.size(); i++) {
             Object value = values.get(i);
-            data[i + 1] = columnValueToString(value);
+            data[i + 1] = evaluateValueToString(value, settings);
         }
         return data;
     }
     
+    private Object formatTooltip(Long value, Object threshold, String data) {
+        List<DataColumn> columns = dataSet.getColumns();
```

Comment:
Hi @jesuino this code is a little twisted for my old mind.

I would do a (guard clause)[https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html] for size < 1.

And then follow teh current logic.

Reply:
@ederign I improved it a little, please let me know if I can improve it more